edb/pgsql/compiler/relgen.py

```diff
@@ -349,29 +351,38 @@ def _get_set_rvar(
                 else:
                     rvars = process_set_as_enumerate(ir_set, stmt, ctx=ctx)
 
-            elif str(expr.func_shortname) == 'std::assert_single':
+            elif fname == 'std::assert_single':
                 # Cardinality assertion (upper bound)
                 rvars = process_set_as_singleton_assertion(
                     ir_set, stmt, ctx=ctx)
 
-            elif str(expr.func_shortname) == 'std::assert_exists':
+            elif fname == 'std::assert_exists':
                 # Cardinality assertion (lower bound)
                 rvars = process_set_as_existence_assertion(
                     ir_set, stmt, ctx=ctx)
 
-            elif str(expr.func_shortname) == 'std::assert_distinct':
+            elif fname == 'std::assert_distinct':
                 # Multiplicity assertion
                 rvars = process_set_as_multiplicity_assertion(
                     ir_set, stmt, ctx=ctx)
 
-            elif str(expr.func_shortname) == 'std::min' and expr.func_sql_expr:
+            elif fname == 'std::min' and expr.func_sql_expr:
                 # Generic std::min
                 rvars = process_set_as_std_min_max(ir_set, stmt, ctx=ctx)
 
-            elif str(expr.func_shortname) == 'std::max' and expr.func_sql_expr:
+            elif fname == 'std::max' and expr.func_sql_expr:
                 # Generic std::max
                 rvars = process_set_as_std_min_max(ir_set, stmt, ctx=ctx)
 
+            elif fname == 'std::range' and expr.func_sql_expr:
+                # Generic std::range() constructor
+                rvars = process_set_as_std_range(ir_set, stmt, ctx=ctx)
+
+            elif fname == 'std::range_of_empty' and expr.func_sql_expr:
```

Comment:
Is this `std::range_of_empty` an artefact of the previous implementation that's no longer relevant? I don't see that function in `std`.

Reply:
Yeah, forgot to clean that up.